packages/core/src/electron-main/electron-main-application-module.ts

```diff
@@ -44,6 +44,7 @@ export default new ContainerModule(bind => {
     bindContributionProvider(bind, ElectronMainApplicationContribution);
 
     bind(ElectronMainApplicationContribution).toService(ElectronMessagingContribution);
+    bind(ElectronMainApplicationContribution).to(TheiaMainApi).inSingletonScope();
```

Comment:
Can Theia use this instead? It's easier to customize downstream.

```js
bind(TheiaMainApi).toSelf().inSingletonScope();
bind(ElectronMainApplicationContribution).toService(TheiaMainApi);
```

Same for the filesystem API binding.

Reply:
Yeah, extricating contributions is a pain.